app/services/telegram_core.py

```
@@ -447,44 +447,43 @@ async def _route_natural_language_fallback(self, chat_id: str, user_id: str, tex
+                "role": "assistant",
                 "content": response_text,
-                "timestamp": datetime.utcnow().isoformat()
+                "timestamp": datetime.utcnow().isoformat(),
+                "intent": metadata.get("intent"),
             })
             self.conversation_contexts[user_id] = context
-            
-            return {"success": True, "response": "AI conversation handled"}
-            
+
+            return {"success": True, "response": "AI conversation handled", "metadata": metadata}
```

Ruff (0.13.2)

Consider moving this statement to an `else` block

(TRY300)